.github/workflows/test.yml

```diff
@@ -0,0 +1,103 @@
+name: Tests
+on:
+  push:
+    paths-ignore:
+      - 'README.md'
+      - 'CHANGELOG.md'
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+          cache: true
+
+      - name: Get dependencies
+        run: go mod download
```

Comment:
Is this step actually required? Doesn't `setup-go` download the module dependencies already?

Reply:
`setup-go` caches the downloaded module but it doesn't download the module I think!
The official doc also installs modules separately.
https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go#building-and-testing-your-code